doc: Rework the AKS tabs so that only instructions for BYOCNI remain. (#28933)
cilium-team/cilium: Documentation/gettingstarted/k8s-install-default.rst

           cilium install |CHART_VERSION|

    .. group-tab:: AKS
       
       .. note::

          On AKS, Cilium can be installed either manually by administrators (Bring your own CNI) or 

>           automatically by AKS (Azure CNI Powered by Cilium). Bring your own CNI offers more flexibility 

Reviewer: Can you remove the use of parenthesis here? Parenthesis are great for adding side-comments, but in this case I think it would make more sense to have the names of the different installation methods outside of parenthesis.

>           and customization as administrators have full control over the installation, but does not 

Reviewer: Grammar suggestion:

```suggestion
          and customization as administrators have full control over the installation, but it does not 
```

Author: i addressed this

          integrate natively with the Azure network stack and administrators need to handle Cilium upgrades. 
          Azure CNI Powered by Cilium integrates natively with the Azure network stack and upgrades are 

>           handled by AKS, but does not offer as much flexibility and customization as it is controlled by AKS. 

Reviewer: Grammar suggestion:

```suggestion
          handled by AKS, but it does not offer as much flexibility and customization as it is controlled by AKS. 
```

Author: addressed this

          The following instructions assume Bring your own CNI, please refer to the Azure CNI Powered by 

>           Cilium external installer guide :ref:`aks_install` for dedicated instructions.

Reviewer 2: This `:ref:` seems to be a dead link?

Author: it depends on this PR https://github.com/cilium/cilium/pull/28286 .. its defined in one of the files added in that PR

>           The following instructions assume Bring your own CNI, please refer to the Azure CNI Powered by 
>           Cilium external installer guide :ref:`aks_install` for dedicated instructions.

Reviewer 2: Minor grammar tidyup:
```suggestion
          The following instructions assume Bring your own CNI. For Azure CNI Powered by 
          Cilium, see the external installer guide :ref:`aks_install` for dedicated instructions.
```

However, the `:ref:` token may substitute a doc title here so you'll want to test out the docs locally and check how the sentence looks.

>        .. include:: ../installation/requirements-aks.rst

Reviewer: Why move the include statement here? I think it makes sense to keep it under the `Install Cilium` header.

Author: i removed legacy ipam and made instructions only for byocni remain as per this discussion https://github.com/cilium/cilium/pull/27199#pullrequestreview-1561391819 .. so no need for separate file and everything can be moved to requirements-aks.rst as like eks/gke

Reviewer: Ah ok that makes sense, thanks!
